producer/buffer/buffer.go

```diff
@@ -85,11 +87,20 @@ func NewBuffer(opts Options) producer.Buffer {
 	if opts == nil {
 		opts = NewOptions()
 	}
+	var (
+		maxMessageSize = opts.MaxMessageSize()
+		maxBufferSize  = opts.MaxBufferSize()
+	)
+	if maxMessageSize > maxBufferSize {
```

Comment:
This seems something that should be caught when setting up the options through validation?

Reply:
sure, can move into a validation function